### PR TITLE
Revised out parameter modifier for C# 7

### LINKED ISSUE #1378: C# Language reference keywords: out
Out variables can be declared where they are first used rather than on a separate statement, starting in C# 7.

### PR DESCRIPTION
**Pull request by rpetrusha**
#Revised out parameter modifier for C# 7

## Summary

Updated the topic to cover C# 7, and also did some cleanup (described in the Details section).

Fixes #1378 

## Details

In addition to documenting the declaration of the `out` variable in the method call, I've done the following:

- Moved code from the `codesnippets` subdirectory to a subdirectory of `samples/snippets`.
- Deleted code in the `codesnippets` subdirectory.
- Removed most of the text for `out.md`, so that the topic only links to the two contextual `out` keywords. The `out-parameter-modifier.md` topic completely duplicated the `out.md` topic, which poses a maintenance nightmare and makes it certain that the identical topics (which also had examples in independent code files) will eventually get out of sync.
- Modified the toc (`toc.md`) so that `out.md` is not in the TOC. It serves to disambiguate `out`, but does not otherwise appear in the TOC.

## Suggested Reviewers

@BillWagner

### CODE REVIEW

File: docs/csharp/language-reference/keywords/out-parameter-modifier.md [hunk 1]
```diff
@@ -53,12 +64,29 @@ The `out` keyword causes arguments to be passed by reference. This is like the [
 -   Async methods, which you define by using the [async](../../../csharp/language-reference/keywords/async.md) modifier.  
   
 -   Iterator methods, which include a [yield return](../../../csharp/language-reference/keywords/yield.md) or `yield break` statement.  
-  
-## Example  
+
+## Declaring an `out` method
```
Comment (BillWagner) [inline docs/csharp/language-reference/keywords/out-parameter-modifier.md]: It's not really an `out` method. It's a method with `out` arguments. Maybe 

## Declaring `out` arguments

is a better heading?

File: docs/csharp/language-reference/keywords/out-parameter-modifier.md [hunk 2]
```diff
@@ -53,12 +64,29 @@ The `out` keyword causes arguments to be passed by reference. This is like the [
...
+
+## Declaring an `out` method   
+
  Declaring an `out` method is useful when you want a method to return multiple values. The following example uses `out` to return three variables with a single method call. Note that the third argument is assigned to null. This enables methods to return values optionally.  
   
- [!code-cs[csrefKeywordsMethodParams#4](../../../csharp/language-reference/keywords/codesnippet/CSharp/out-parameter-modifier_4.cs)]  
-  
+ [!code-cs[csrefKeywordsMethodParams#4](../../../../samples/snippets/csharp/language-reference/keywords/out/out-4.cs)]  
+
+ A common pattern involves returning a `bool` to indicate whether an operation succeeded and failed, and returning the value produced by the operation in an `out` argument. A number of parsing methods, such as the @System.DateTime.TryParse(System.String,@System.DateTime) method, use this pattern.
```
Comment (BillWagner) [inline docs/csharp/language-reference/keywords/out-parameter-modifier.md]: We're trying to standardize on calling this the "Try* Pattern". It'd be great to use that term here.

**Review by BillWagner — APPROVED** [review]
@rpetrusha I like the changes and the reorganization.

I left a couple suggestions for possible changes. I'm fine merging this either way. :shipit:

**Comment by rpetrusha** [comment]
Thanks, @BillWagner. I've incorporated both of your suggestions. I'll merge as soon as the topic builds.

**Pull request merged**